docs/integrations-and-plugins/pandera.md

```diff
@@ -0,0 +1,687 @@
+
+    For nullable integer columns, use `float` type instead of `int` because pandas represents nullable integers as float64 (since NaN is a float value).
+
+
+
+Next, create a hook to validate datasets. Create `src/spaceflights_pandera/hooks/validation.py`:
+
+```python
+"""Kedro hooks for Pandera validation integration."""
+import logging
+from typing import Any, Dict
+
+import pandas as pd
+from kedro.framework.hooks import hook_impl
+from kedro.io import DataCatalog
+from pandera.errors import SchemaError, SchemaErrors
+
+from ..schemas.raw import companies_schema, shuttles_schema, reviews_schema
+
+logger = logging.getLogger(__name__)
+
+
+class PanderaValidationHooks:
+    """Kedro hooks that integrate Pandera validation."""
+
+    def __init__(self):
+        # Map catalog dataset names to their validation schemas
+        self.raw_schemas = {
+            "companies": companies_schema,
+            "shuttles": shuttles_schema,
+            "reviews": reviews_schema,
+        }
+
+    @hook_impl
+    def before_node_run(
+        self,
+        node,
+        catalog: DataCatalog,
+        inputs: Dict[str, Any],
+        is_async: bool,
+    ) -> Dict[str, Any]:
+        """Validate and coerce input data before node execution."""
+        validated_inputs = {}
+
+        for input_name, data in inputs.items():
+            schema = self.raw_schemas.get(input_name)
+
+            if schema is None or not isinstance(data, pd.DataFrame):
+                validated_inputs[input_name] = data
+                continue
+
+            logger.info(f"Validating '{input_name}' for node '{node.name}'")
+
+            try:
+                validated_data = schema.validate(data, lazy=True)
+                logger.info(f"✓ Validation passed for '{input_name}'")
+                validated_inputs[input_name] = validated_data
+            except (SchemaError, SchemaErrors) as e:
+                logger.error("Validation failed for '%s': %s", input_name, e)
+                raise
+
+        return validated_inputs
+```
+
+Finally, register the hook in `src/spaceflights_pandera/settings.py`:
+
+```python
+"""Project settings."""
+from spaceflights_pandera.hooks.validation import PanderaValidationHooks
+
+HOOKS = (PanderaValidationHooks(),)
+```
+
+From this point, when you execute `kedro run` you will see the validation logs:
+
+```console
+[10/01/25 11:10:13] INFO     Validating 'companies' for node 'preprocess_companies_node'
+                    INFO     ✓ Validation passed for 'companies'
+                    INFO     Validating 'shuttles' for node 'preprocess_shuttles_node'
+                    INFO     ✓ Validation passed for 'shuttles'
+                    INFO     Validating 'reviews' for node 'create_model_input_table_node'
+                    INFO     ✓ Validation passed for 'reviews'
+```
+
+### What happens when validation fails?
+
+Let's make one schema rule deliberately too strict so you can see Pandera in action.
```

Comment:
```suggestion
Let's make one schema rule deliberately fail so you can see Pandera in action.
```
Nit: I think "too strict" would be more like if it was the wrong type but still usable; arbitrarily high value sounds more like deliberately failing. :)